### PR TITLE
Support yolov5n

### PR DESCRIPTION
**Pull request by ChloeWu1**
Signed-off-by: Fan, Tao [tao.fan@intel.com](mailto:tao.fan@intel.com) , Wu, Jiawei [jiawei.wu@intel.com](mailto:jiawei.wu@intel.com)

### CODE REVIEW

File: dynamic_vino_lib/src/models/object_detection_yolov5_model.cpp [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    convert_element_type(ov::element::f32).
+    convert_color(ov::preprocess::ColorFormat::RGB).scale({255., 255., 255.});
+  ppp.input().model().set_layout("NCHW");
+  addInputInfo("input", input_tensor_name_);
+
+  // set output property
+  auto output_info_map = model->outputs();
+  if (output_info_map.size() != 1) {
+    slog::warn << "This model seems not Yolo-like! We got "
+      << std::to_string(output_info_map.size()) << "outputs, but SSDnet has only one."
```
[the remark this answers is not in the document]
Comment (LewisLiuPub) [reply dynamic_vino_lib/src/models/object_detection_yolov5_model.cpp]: Fixed.

File: dynamic_vino_lib/src/models/object_detection_yolov5_model.cpp [hunk 2]
```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2018 Intel Corporation
```
[the remark this answers is not in the document]
Comment (ChloeWu1) [reply dynamic_vino_lib/src/models/object_detection_yolov5_model.cpp]: Fixed.

File: dynamic_vino_lib/src/models/object_detection_yolov5_model.cpp [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/** 
+ * @brief a header file with declaration of ObjectDetectionModel class
+ * @file object_detection_yolov5_model.cpp
+ */
+
+#include "dynamic_vino_lib/models/object_detection_yolov5_model.hpp"
```
[the remark this answers is not in the document]
Comment (ChloeWu1) [reply dynamic_vino_lib/src/models/object_detection_yolov5_model.cpp]: Fixed.

File: dynamic_vino_lib/src/models/object_detection_yolov5_model.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (ChloeWu1) [reply dynamic_vino_lib/src/models/object_detection_yolov5_model.cpp]: Fixed.

File: dynamic_vino_lib/src/models/object_detection_yolov5_model.cpp [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        auto y_min = y - (h / 2);
+
+        boxes.emplace_back(x_min, y_min, w, h);
+    }
+  }
+
+  std::vector<int> nms_result;
+  cv::dnn::NMSBoxes(boxes, confidences, confidence_thresh, NMS_THRESHOLD, nms_result);
+  for (int idx: nms_result) {
+      if (class_ids[idx] != 0)
```
[the remark this answers is not in the document]
Comment (ChloeWu1) [reply dynamic_vino_lib/src/models/object_detection_yolov5_model.cpp]: Fixed.

File: dynamic_vino_lib/src/models/object_detection_yolov5_model.cpp [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+  setFrameSize(frame.cols, frame.rows);
+
+  if (!matToBlob(frame, input_frame_loc, 1, 0, engine)) {
+    return false;
+  }
+  return true;
+}
+
+
+bool Models::ObjectDetectionYolov5Model::matToBlob(
```
[the remark this answers is not in the document]
Comment (LewisLiuPub) [reply dynamic_vino_lib/src/models/object_detection_yolov5_model.cpp:111]: Ignore this comment. The matToBlob is already implemented with new tensor method.

File: dynamic_vino_lib/src/models/object_detection_yolov5_model.cpp [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+      std::string label = class_ids[idx] < labels.size() ?
+        labels[class_ids[idx]] : std::string("label #") + std::to_string(class_ids[idx]);
+      result.setLabel(label);
+      results.push_back(result);
+  }
+
+  return true;
+}
+
+Models::Resize_t Models::ObjectDetectionYolov5Model::pre_process_ov(const cv::Mat &input_image) {
```
[the remark this answers is not in the document]
Comment (ChloeWu1) [reply dynamic_vino_lib/src/models/object_detection_yolov5_model.cpp]: Fixed.

[the remark this answers is not in the document]
Comment (LewisLiuPub) [reply dynamic_vino_lib/src/models/object_detection_yolov5_model.cpp]: double check.
If the resizing step is really necessary, please revert to the previous solution. sorry for the confusion.

[the remark this answers is not in the document]
Comment (ChloeWu1) [reply dynamic_vino_lib/src/models/object_detection_yolov5_model.cpp:198]: Reverted to the previous solution.

**Pull request merged**